### ISSUE TITLE
Tests

### ISSUE THREAD

**Issue by Fogapod** [IOMirea/rpc#7]
Tests using mock, maybe tests involving real redis in travis, coverage

**Comment by Physsix27** [comment]
Can you assign this issue to me, I just started working on it

**Comment by Fogapod** [comment]
Alright, it would be very helpful if you could do this

**Comment by Fogapod** [comment]
I apoligize for the inconveniences, but several breaking changes had to be made. All changes are reflected in existing examples.
Changes:
- `Server.respond` -> `Request.reply`
- Removed `Server` argument in callbacks. It can be accessed using `Request.server`
- Removed `flatten` method of responses. It was replaced with plain `__await__`

**Comment by Physsix27** [comment]
Were these changes made already? are they made into the master branch, if they are I can definitely pull and it shouldn't be a problem.

I am currently working on getting everything set up, the coverage page for the test and all the `.py` files. and then starting to unit test one by one python file.

**Comment by Fogapod** [comment]
Yes, they are in master

**Comment by Physsix27** [comment]
I am having some issue working with Travis CI since is my first time, I am getting an error for in the virtualenv for Travis that is not recognizing one of my import. 

`___________ ERROR collecting tests/unit/enums_py/StatusCode_test.py ____________
ImportError while importing test module '/home/travis/build/IOMirea/yarpc/tests/unit/enums_py/StatusCode_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
tests/unit/enums_py/StatusCode_test.py:3: in <module>
    from iomirea_rpc import enums
E   ModuleNotFoundError: No module named 'iomirea_rpc'`

**Comment by Fogapod** [comment]
Sorry for that, library was renamed to `yarpc` recently

**Comment by Physsix27** [comment]
No worries I restarted and using pytests as you want it

**Comment by Physsix27** [comment]
Hey, let me know if you feel this my PR #22  is missing anything to resolve thIS issue.

**Comment by Fogapod** [comment]
@Physsix27 are you going to write tests or you only wanted to configure make, Travis?

**Comment by Physsix27** [comment]
Yes I want to continue writing tests but right now I am not going to be able to bring the project to 100% coverage, I can just write test for some python files. I can write tests for `request.py`, `reponse.py` and `abc.py` and bring those 3 files up to 100% coverage.
This is how they are currently looking:
![image](https://user-images.githubusercontent.com/36493537/66708139-e1d04980-ed19-11e9-8a86-8768c3db055e.png)

**Comment by Physsix27** [comment]
I will work on the tests later today for the files I mentioned above

**Comment by Gelbpunkt** [comment]
I'd suggest testing against a real redis install via docker in Travis.

**Comment by Physsix27** [comment]
@Gelbpunkt Can I get a little more detail about that and what is that since I have never done that?

**Comment by Gelbpunkt** [comment]
@Fogapod does not like my idea, I think.
Docker is a container daemon that runs on almost all platforms. It runs stuff mostly isolated from your OS. You can for example run Ubuntu or Alpine Linux in a container on a Mac. Inside a container, you can have an image. Those can be a basic OS or even contain software, for example Redis.

https://hub.docker.com/_/redis

You'd run `docker pull redis:5.0.6-alpine` to pull the image to your PC (using Alpine Linux has smaller images usually).
Then you can run it and map the port inside the container to your PC.
`docker run --rm -d --name redis -p 6739:6739 redis:5.0.6-alpine`
This will start a container in the background which runs Redis on the container, mapping the port to your PC.

You can then use redis-cli on your machine to connect to the redis in the container.

Containers are platform-independent (mostly), usually very up to date, depending on the image, and extremely secure due to being mostly separate from the host.

Hope that helps ;)

**Comment by Physsix27** [comment]
I don't understand what you want me to do, which looks like a different issue which I don't understand much and would have to do some research to learn more about that topic. I want it to do more python tests because that is the skill I am currently trying to improve. but what you are explaining seems a little different and does not specifies what is it that I need to test. `Testing against a real redis` I am not sure what that means. @Gelbpunkt

**Comment by Gelbpunkt** [comment]
I meant that we should maybe think about not using mocks but instead using a Redis with client and server to test them

**Comment by Physsix27** [comment]
@Gelbpunkt I understand what you are trying to say but you still need to test the python code of the project as well no matter which path you take, that would be a decision of the project owner to take

**Comment by Fogapod** [comment]
@Physsix27 Is there any more progress on this?
I should inform you that project was renamed from yarpc to jarpc. I hope this won't happen anymore
